### ISSUE TITLE
FILE_PASS is not expanding the secret file's content

### ISSUE THREAD

**Issue by Xav-Pe** [linuxserver/docker-transmission#129]
Hello,

The doc says you can use ``FILE__PASSWORD`` to use password from a file. Here is what I did with docker-compose : 

```
services:
  transmission:
    container_name: transmission
    image: linuxserver/transmission
    environment:
[...]
      USER: 'myuser'
      FILE__PASSWORD: '/run/secrets/transmission-pass'
[...]
    secrets:
      - transmission-pass
[...]
secrets:
  transmission-pass:
    file: ./transmission-pass.txt
```

When running ``docker-compose up``, the output says : ``[env-init] PASSWORD set from FILE__PASSWORD``. But, from inside the container, the ``FILE__PASSWORD`` contains the path to the secret, not the content of it : 

```
root@c2a6a3f6b3f6:/# env | grep PASS
FILE__PASSWORD=/run/secrets/transmission-pass
root@c2a6a3f6b3f6:/# cat /run/secrets/transmission-pass
mypassword
```

And the webUI doesn't ask for a login/password.

When using ``FILE__PASS`` instead ``FILE__PASSWORD``, the UI asks for a login/password, but the password has to be left blank.

I don't know if the environment variable content should be the password, or if transmission is just not reading the file's content.

## Expected Behavior
The ``FILE__PASS`` (or ``FILE__PASSWORD``?) should be read and used for the UI password.

## Current Behavior
The ``FILE__PASS`` (or ``FILE__PASSWORD``?) environment variable is filled with the path to the secret.

EDIT : maybe the content should not be in the environment variable, but anyway the password is not set.

## Steps to Reproduce
Create a docker-compose with these parts (complete with your environment)

```
services:
  transmission:
    container_name: transmission
    image: linuxserver/transmission
    environment:
[...]
      USER: 'myuser'
      FILE__PASSWORD: '/run/secrets/transmission-pass'
[...]
    secrets:
      - transmission-pass
[...]
secrets:
  transmission-pass:
    file: ./transmission-pass.txt
```

run it with ``docker-compose up``, log into the container with ``docker exec -it transmission bash``, then look at the environment variables ``env | grep PASS``.

## Environment
**OS:**      Debian 10.6
**CPU architecture:** x86_64
**How docker service was installed:**  package. ``Docker version 19.03.12, build 48a66213fe``    / ``docker-compose version 1.26.2, build eefe0d31``.

## Docker logs
```
Creating network "transmission_transmission-lan" with driver "bridge"
Creating transmission ... done
Attaching to transmission
transmission    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
transmission    | [s6-init] ensuring user provided files have correct perms...exited 0.
transmission    | [fix-attrs.d] applying ownership & permissions fixes...
transmission    | [fix-attrs.d] done.
transmission    | [cont-init.d] executing container initialization scripts...
transmission    | [cont-init.d] 01-envfile: executing...
transmission    | [env-init] PASS set from FILE__PASS
transmission    | [cont-init.d] 01-envfile: exited 0.
transmission    | [cont-init.d] 10-adduser: executing...
transmission    |
transmission    | -------------------------------------
transmission    |           _         ()
transmission    |          | |  ___   _    __
transmission    |          | | / __| | |  /  \
transmission    |          | | \__ \ | | | () |
transmission    |          |_| |___/ |_|  \__/
transmission    |
transmission    |
transmission    | Brought to you by linuxserver.io
transmission    | -------------------------------------
transmission    |
transmission    | To support LSIO projects visit:
transmission    | https://www.linuxserver.io/donate/
transmission    | -------------------------------------
transmission    | GID/UID
transmission    | -------------------------------------
transmission    |
transmission    | User uid:    1000
transmission    | User gid:    1000
transmission    | -------------------------------------
transmission    |
transmission    | [cont-init.d] 10-adduser: exited 0.
transmission    | [cont-init.d] 20-config: executing...
transmission    | sed: unsupported command "
transmission    | [cont-init.d] 20-config: exited 0.
transmission    | [cont-init.d] 99-custom-files: executing...
transmission    | [custom-init] no custom files found exiting...
transmission    | [cont-init.d] 99-custom-files: exited 0.
transmission    | [cont-init.d] done.
transmission    | [services.d] starting services
transmission    | [services.d] done.
transmission    | [2020-10-03 16:14:01.759] Transmission 3.00 (bb6b5a062e) started (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/session.c:769)
transmission    | [2020-10-03 16:14:01.759] RPC Server Adding address to whitelist: 127.0.0.1 (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:956)
transmission    | [2020-10-03 16:14:01.759] RPC Server Serving RPC and Web requests on 0.0.0.0:9091/transmission/ (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:1243)
transmission    | [2020-10-03 16:14:01.759] RPC Server Password required (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:1254)
transmission    | [2020-10-03 16:14:01.759] DHT Generating new id (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-dht.c:389)
transmission    | [2020-10-03 16:14:01.759] Using settings from "/config" (/home/buildozer/aports/community/transmission/src/transmission-3.00/daemon/daemon.c:646)
transmission    | [2020-10-03 16:14:01.759] Saved "/config/settings.json" (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/variant.c:1221)
transmission    | [2020-10-03 16:14:01.759] transmission-daemon requiring authentication (/home/buildozer/aports/community/transmission/src/transmission-3.00/daemon/daemon.c:674)
transmission    | [2020-10-03 16:14:01.759] Watching "/watch" for new .torrent files (/home/buildozer/aports/community/transmission/src/transmission-3.00/daemon/daemon.c:698)
transmission    | [2020-10-03 16:14:01.759] Port Forwarding (NAT-PMP) initnatpmp succeeded (0) (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/natpmp.c:73)
transmission    | [2020-10-03 16:14:01.759] Port Forwarding (NAT-PMP) sendpublicaddressrequest succeeded (2) (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/natpmp.c:73)
transmission    | [2020-10-03 16:14:09.758] Port Forwarding State changed from "Not forwarded" to "Starting" (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/port-forwarding.c:106)
transmission    | [2020-10-03 16:14:09.758] Port Forwarding State changed from "Starting" to "???" (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/port-forwarding.c:106)
transmission    | [2020-10-03 16:14:21.756] DHT Attempting bootstrap from dht.transmissionbt.com (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-dht.c:299)

```

Thanks!

**Comment by aptalca** [comment]
It looks like you're having two separate issues.
1) To set the var `PASS`, you need to use `FILE__PASS`, not `FILE__PASSWORD`
2) You need to create the secret with printf so there is no trailing line ending

**Comment by Xav-Pe** [comment]
>To set the var PASS, you need to use FILE__PASS, not FILE__PASSWORD

I agree, but the doc says ``FILE__PASSWORD``. Anyway, I tried both : 

>When using FILE__PASS instead FILE__PASSWORD, the UI asks for a login/password, but the password has to be left blank.

For your second point : 

>You need to create the secret with printf so there is no trailing line ending

That was it I thinks. I created my secret file with ``vi``, and it seems it adds a trailing line break... I wasted so much time on it. 

Anyway, I still thinks the README should be updated with ``FILE__PASS``.

Thanks for your help!

**Comment by tobbenb** [comment]
You mistake the example of how to use the FILE__ with the env variable for setting the password for this app.
Below the example it says that it sets the variable PASSWORD, but for this app the variable is PASS. The example is in every README we have.

**Comment by Xav-Pe** [comment]
You're right, thanks. I close this one, sorry!

**Comment by stoinov** [comment]
@aptalca can you give an example of the `printf` command? I am using `printf "%s" "password-string" > /path/to/transmission/password` and then providing `FILE__PASS=/path/to/transmission/password` yet the `rpc-authentication-required` is always false.

**Comment by aptalca** [comment]
The file path needs to be inside the container

**Comment by stoinov** [comment]
Correct. This path is the same as where the config JSON file is.

**Comment by stoinov** [comment]
Just to clarify - I made sure the container is started with the same user ID as the folder and files owner.
When ran with the PASS="password-string" it's properly setting the authentication and I am asked for the exact password.
My pass contains special character that requires putting in quotes, but I also tried using simple password and I still get the same problem.
I am really out of ideas what could be causing the issue.
